pkg/graveler/ref/manager.go

```diff
@@ -163,16 +165,61 @@ func (m *KVManager) ListRepositories(ctx context.Context) (graveler.RepositoryIt
 	return NewKVRepositoryIterator(ctx, &m.kvStore)
 }
 
+func (m *KVManager) updateRepoState(ctx context.Context, repo *graveler.RepositoryRecord, state graveler.RepositoryState) error {
+	repo.State = state
+	return m.kvStore.SetMsg(ctx, graveler.RepositoriesPartition(), []byte(graveler.RepoPath(repo.RepositoryID)), graveler.ProtoFromRepo(repo))
+}
+
+func (m *KVManager) deleteRepository(ctx context.Context, repo *graveler.RepositoryRecord) error {
+	// ctx := context.Background() TODO (niro): When running this async create a new context and remove ctx from signature
+	logger := logging.FromContext(ctx)
+	// Scan through repository partition and delete entities, using SecondaryIndex struct since we don't really need the data, just the key
+	// Failures on deleting repository entities don't constitute as repository deletion failure
+	partition := graveler.RepoPartition(repo)
+	it, err := kv.NewPartitionIterator(ctx, m.kvStore.Store, (&kv.SecondaryIndex{}).ProtoReflect().Type(), partition)
+	if err != nil {
+		logger.WithField("repository", repo.RepositoryID).
```

Comment:
You can start this function with `logger := logger.WithField("repository", repo.RepositoryID)` and use `logger` everywhere instead of creating the same field again and again